src/Foundation/NSUrlSessionHandler.cs

```diff
@@ -53,8 +56,68 @@ namespace System.Net.Http {
 #else
 namespace Foundation {
 #endif
+
+	// useful extensions for the class in order to set it in a header
+	static class NSHttpCookieExtensions
+	{
+		private static bool AppendSegment(StringBuilder builder, bool first, string name, string value)
+		{
+			if (first)
+				first = false;
+			else
+				builder.Append("; ");
+
+			builder.Append(name);
+			if (value != null) {
+				builder.Append("=");
+				builder.Append(value);
+			}
+			return first;
+		}
+
+		// returns the header for a cookie
+		public static string GetHeaderValue (this NSHttpCookie cookie)
+		{
+			var header = new StringBuilder();
+			var first = true;
+			first = AppendSegment (header, first, cookie.Name, cookie.Value);
+			first = AppendSegment (header, first, NSHttpCookie.KeyPath.ToString (), cookie.Path.ToString ());
+			first = AppendSegment (header, first, NSHttpCookie.KeyDomain.ToString (), cookie.Domain.ToString ());
+			first = AppendSegment (header, first, NSHttpCookie.KeyVersion.ToString (), cookie.Version.ToString ());
+
+			if (cookie.Comment != null)
+				first = AppendSegment (header, first, NSHttpCookie.KeyComment.ToString (), cookie.Comment.ToString());
+
+			if (cookie.CommentUrl != null)
+				first = AppendSegment (header, first, NSHttpCookie.KeyCommentUrl.ToString (), cookie.CommentUrl.ToString());
+
+			if (cookie.Properties.ContainsKey (NSHttpCookie.KeyDiscard))
+				first = AppendSegment (header, first, NSHttpCookie.KeyDiscard.ToString (), null);
+
+			if (cookie.ExpiresDate != null) {
+				// Format according to RFC1123; 'r' uses invariant info (DateTimeFormatInfo.InvariantInfo)
+				var dateStr = ((DateTime) cookie.ExpiresDate).ToUniversalTime().ToString("r", CultureInfo.InvariantCulture);
```

Comment:
Style: space before opening parenthesis (happens a few more times in this method as well).